sql/expression/function/sleep.go

```diff
@@ -37,8 +37,12 @@ func (s *Sleep) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		return nil, err
 	}
 
-	time.Sleep(time.Duration(child.(float64)*1000) * time.Millisecond)
-	return 0, nil
+	select {
+	case <-time.After(time.Duration(child.(float64)) * 1000.0):
```

Comment:
So I think this implies that `child` is in microseconds? To make that more self-documenting, I suggest:

```go
time.Duration(child.(float64))*time.Microsecond
```

Reply:
`child` is in seconds but mysql takes decimals so we convert to milliseconds.

Reply:
Right, ok, but `time.After` and `time.Sleep` take nanoseconds (which is what `time.Duration` is), so multiplying by 1000 is definitely not right.

Reply:
Are you sure about that? It works...

Reply:
Oh frag, I notices it doesn't have the `time.Milliseconds` in the version in this PR (I was testing with gitbase so modifiying in another branch, that's why it worked, my version had it).

Reply:
Ah, if it's seconds then you want `time.Duration(child.(float64))*time.Second`, that will give you the right scaling without an extra factor.

Reply:
(Or if you're concerned about fractions, then you probably want the multiplier _before_ the conversion to int: `time.Duration(child.(float64) * 1000) * time.Millisecond` for example).

Reply:
Multiplying by time.seconds seems to floor() it:

```

Reply:
Added the 1000 inside the parents, thanks again!

Reply:
You'll still want the `time.Milliseconds` scaling factor outside the parens, though, after the conversion to `time.Duration`.